src/java/org/apache/cassandra/cql3/statements/ModificationStatement.java

```diff
@@ -1044,4 +1121,23 @@ public List<Pair<ColumnIdentifier, ColumnCondition.Raw>> getConditions()
             return conditions;
         }
     }
+
+    private static final Constants.Value ONE = new Constants.Value(ByteBufferUtil.bytes(1));
+
+    public SelectStatement createSelectForTxn()
+    {
+        // TODO: get working with static-only updates that don't specify any/all primary key columns
+        Preconditions.checkState(getRestrictions().hasAllPKColumnsRestrictedByEqualities());
+        Selection selection = Selection.forColumns(metadata, Lists.newArrayList(requiresRead), false);
```

Comment:
`requiresRead` is computed in the constructor, which is before migration... but if a Substitutions already exists, then won't that column be missing? `org.apache.cassandra.cql3.statements.UpdateStatement.ParsedInsert#prepareInternal` moves them there first.

Reply:
feedback branch shows this is an issue